web_src/js/htmx.js

```diff
@@ -19,3 +20,15 @@ document.body.addEventListener('htmx:responseError', (event) => {
   // TODO: add translations
   showErrorToast(`Error ${event.detail.xhr.status} when calling ${event.detail.requestConfig.path}`);
 });
+
+// eslint-disable-next-line no-import-assign
+htmx.createWebSocket = (url) => {
+  // TODO: reuse websocket from shared webworker
+  const sock = new WebSocket(url, []);
+  sock.binaryType = htmx.config.wsBinaryType;
+  return sock;
+};
```

Comment:
```suggestion
let webSocket;

// TODO: move websocket creation to shared webworker
htmx.createWebSocket = (url) => { // eslint-disable-next-line no-import-assign
  if (![0, 1].includes(webSocket?.readyState)) return webSocket;
  const ws = new WebSocket(url, []);
  ws.binaryType = htmx.config.wsBinaryType;
  webSocket = ws;
  return ws;
};
```
This should be enough for basic re-use. If the websocket is not connecting or open, it will create a new one.

Reply:
This code would re-use the same websocket for new connections ignoring different urls?! 

As I wrote before I am still not sure if I get why you want this change. It would make sense to me as soon as we would re-use sockets across tabs with sth like a shared webworker, but with your suggestion it will allow us to re-use the socket  when we would add multiple `ws-connect="/-/ws"` to different dom elements. Why would we need that? One `ws-connect` per tab should be totally sufficient as you can add, update or remove any dom element through it.

Reply:
My only concern was multiple `ws-connect` on the same page but I see you moved it to `body` now and body is singluar so it's fine to not do this suggestion.